ContextFactory/ConfiguredContextFactory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace JMS\SerializerBundle\ContextFactory;
+
+use JMS\Serializer\Context;
+use JMS\Serializer\ContextFactory\DeserializationContextFactoryInterface;
+use JMS\Serializer\ContextFactory\SerializationContextFactoryInterface;
+use JMS\Serializer\DeserializationContext;
+use JMS\Serializer\SerializationContext;
+
+/**
+ * Class ConfiguredContextFactory
+ */
+class ConfiguredContextFactory implements SerializationContextFactoryInterface, DeserializationContextFactoryInterface
+{
+    /**
+     * Context config
+     *
+     * @var array
+     */
+    private $config;
+
+    /**
+     * ConfiguredContextFactory constructor.
+     *
+     * @param array $config Context configuration
+     */
+    public function __construct(array $config)
```

Comment:
a factory is meant to be configured. what should not change is the context. 
you can configure the service via `JMSSerializerExtension`